### PR TITLE
[WIP-donotmerge] List Objects version 2.

### PR DESCRIPTION
**Pull request by krishnasrinivas**

### CODE REVIEW

File: api-resources.go [hunk 1]
```diff
@@ -17,12 +17,69 @@
...
+	}
+	delimiter = continuationToken.Delimiter
+	marker = continuationToken.Marker
+	return
+}
+
+// Parse bucket url queries
+func getBucketResources(values url.Values) (listType int, prefix, marker, delimiter string, maxkeys int, encodingType string) {
+	if values.Get("list-type") != "" {
+		listType, _ = strconv.Atoi(values.Get("list-type"))
```
Comment (harshavardhana) [inline api-resources.go:81]: we can errorIf her and fail later with listType == 0

**Comment by harshavardhana** [comment]
@krishnasrinivas also can you add minimal unit tests with this patch as well ?

**Comment by harshavardhana** [comment]
Is this still a work in progress?

**Comment by krishnasrinivas** [comment]
@harshavardhana it's still WIP

**Pull request closed without merging**